### PR TITLE
Add NodeLink voice receiver & HTTP compression (Brotli)

### PR DESCRIPTION
**Pull request by ThePedroo**
This PR has the focus to add NodeLink-specific features to moonlink.js, allowing NodeLink users to use features like:

- Voice receiving
- HTTP compression enabled by default (applicable with LavaLink as well, through editing `application.yml` file)

Introducing `listenVoice` and `stopListeningVoice` methods to Player, it allows the developer using moonlink.js to record and/or use it for bots that uses Text-To-Speech, allowing it to smoothly listen while still playing.

### CODE REVIEW

**Review by 1Lucas1apk — COMMENTED** [review]
You could also use `${this.node.address}` 
Instead of `${this.node.host}:${this.node.port}`

File: src/@Entities/MoonlinkPlayer.ts [hunk 1]
```diff
@@ -553,4 +562,74 @@ export class MoonlinkPlayer {
...
+    /**
+     * Returns an event listener containing voice data received from the NodeLink server.
+     * @note NodeLink only feature
+     * @returns An event listener that emits "start", "stop", "close", "error", and "unknown" events.
+     */
+    public listenVoice(): EventEmitter | boolean {
+        if (!this.node.info.isNodeLink)
+          return false;
+
+        this.voiceReceiverWs = new MoonlinkWebSocket(`ws${this.node.secure ? "s" : ""}://${this.node.host}:${this.node.port}/connection/data`, {
```
Comment (1Lucas1apk) [inline src/@Entities/MoonlinkPlayer.ts]: In this part

**Comment by ThePedroo** [comment]
AFAIK `node.address` includes `/v4` endpoint prefix which isn't in NodeLink voice receiving endpoint, which is `/connection/data`. I'll review further once I have time

**Comment by 1Lucas1apk** [comment]
> AFAIK `node.address` includes `/v4` endpoint prefix which isn't in NodeLink voice receiving endpoint, which is `/connection/data`. I'll review further once I have time

https://github.com/Ecliptia/moonlink.js/blob/v3/src%2F%40Entities%2FMoonlinkNode.ts#L70-L72

It returns the same thing as this.node.host:this.node.port

**Comment by ThePedroo** [comment]
Ah, the method. I'll update once possible, thanks

**Comment by 1Lucas1apk** [comment]
All you need to do is compile the code

**Comment by ThePedroo** [comment]
Done. For further future compability, I highly recommend following [NodeLink documentation](https://github.com/PerformanC/NodeLink/blob/v2.0.0/docs/API.md), however I'll try to maintain this implementation. Thank you

**Review by 1Lucas1apk — APPROVED** [review]

**Pull request merged**